packages/block-editor/src/components/grid/use-grid-layout-sync.js

```diff
@@ -11,6 +11,49 @@ import { usePrevious } from '@wordpress/compose';
 import { store as blockEditorStore } from '../../store';
 import { GridRect } from './utils';
 
+/**
+ * Returns an attributes 'update' object for unsetting columnStart and rowStart
+ * `styles.layout` attributes.
+ *
+ * Will return `undefined` when the attributes are already not set indicating
+ * there's nothing to do.
+ *
+ * @param {Object}  attributes          The current block attributes.
+ * @param {boolean} unsetSpanAttributes Whether rowSpan and columnSpans should
+ *                                      also be unset.
+ *
+ * @return {undefined|Object} The attribute updates or `undefined` when the
+ *                            attribute are already not set.
+ */
+function unsetGridPositionAttributes(
+	attributes,
+	unsetSpanAttributes = false
+) {
+	let layout;
+	let hasUpdate;
+
+	if ( unsetSpanAttributes ) {
+		const { columnStart, rowStart, columnSpan, rowSpan, ...restLayout } =
+			attributes.style?.layout ?? {};
+		layout = restLayout;
+		hasUpdate = !! columnStart || rowStart || columnSpan || rowSpan;
+	} else {
+		const { columnStart, rowStart, ...restLayout } =
+			attributes.style?.layout ?? {};
+		layout = restLayout;
+		hasUpdate = !! columnStart || rowStart;
+	}
+
+	if ( hasUpdate ) {
+		if ( ! Object.keys( layout ).length ) {
+			return { style: { ...attributes.style, layout: undefined } };
+		}
+		return {
+			style: { ...attributes.style, layout },
+		};
+	}
+}
```

Comment:
Ugh, this started off simple, but got more complicated as things developed. I wonder if it's worth unDRYing it.

Reply:
Yeah I don't think there's a lot of value to having a helper function here given the need for `unsetSpanAttributes`.

If you're wanting to clean it up, maybe try using `setImmutably` to set the attributes to `undefined` instead of destructuring. That might read a little bit nicer. I'm pretty sure we can safely set attributes to `undefined` to remove them because `serialize()` will strip them out.

Reply:
Yep, I originally tried with setting them to `undefined` so it does work 👍 

I'll try some refactorings.

Reply:
I've updated it to use `setImmutably`, which is ok, though it's not so great with multiple attributes.

One negative is that it leaves behind an empty `layout` object after everything has been unset inside it. I feel like it's adding too much complexity when I try to handle that, so maybe it's ok to leave it. 🤔

Reply:
Pushed one more update which I think is better, solves the empty `layout` object issue.

Still have the `style` object which can be empty, but at least that's more likely to be used outside of grid.

Reply:
Maybe we need `unsetImmutably` which handles removing dangling objects.

_quiet voice_ I miss lodash...